src/natcap/invest/validation.py

```diff
@@ -193,9 +193,26 @@ def check_directory(dirpath, must_exist=True, permissions='rx', **kwargs):
                 dirpath = parent
                 break
 
-    permissions_warning = check_permissions(dirpath, permissions, True)
-    if permissions_warning:
-        return permissions_warning
+    MESSAGE_KEY = 'NEED_PERMISSION_DIRECTORY'
+
+    if 'r' in permissions:
+        try:
+            os.scandir(dirpath).close()
+        except:
+            return MESSAGES[MESSAGE_KEY].format(permission='read')
+
+    # Check for x access before checking for w, since w operations to a dir are dependent on x access
+    if 'x' in permissions:
+        try:
+            os.chdir(dirpath)
+        except:
+            return MESSAGES[MESSAGE_KEY].format(permission='execute')
+
+    if 'w' in permissions:
+        try:
+            tempfile.TemporaryFile(dir=dirpath)
```

Comment:
@emilyanndavis does this need to be used a context manager: `with tempfile.TemporaryFile(dir=dirpath)`? Otherwise this might be holding open a connection to the file and causing the Permission errors in the tests?

Also , do we need to be sure to remove the tempfile after creating it?

Reply:
> Also , do we need to be sure to remove the tempfile after creating it?
I think you answered this on Slack, and it's answered in the docs, `TemporaryFile` and `TemporaryDirectory` should be cleaning up after themselves.

Reply:
Your instinct was right, though. Without a context manager, I do need to close the file to ensure it is cleaned up. I updated this locally, and I think it fixed some issues, but not all of them.